code/datums/action.dm

```diff
@@ -552,6 +552,23 @@
 	owner.forceMove(box)
 	owner.playsound_local(box, 'sound/misc/box_deploy.ogg', 50, TRUE)
 
+/datum/action/item_action/nanoforgedkatana
+	name = "Blood cleaver"
+	desc = "Swing your blade in a wide arc, hitting all enemies in a 3x2 rectangle in the click direction"
+	//icon here
+	background_icon_state = "bg_agent"
+	check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUN|AB_CHECK_CONSCIOUS
+	COOLDOWN_DECLARE(cleaver_cooldown)
+
+/datum/action/item_action/nanoforgedkatana/Trigger()
+	if(!isturf(owner.loc))
+		to_chat(owner, "<span class = 'notice'>You cannot execute such a wide slash in such a small space!</span>")
+		return
+	if(!COOLDOWN_FINISHED(src, cleaver_cooldown))
+		return
```

Comment:
So on discord you raised an issue of the overlay issue.

One option is to make it so when they use it before cooldown ends, they get a message directly to them saying how long is left.

IIRC there's also a way to have procs call when a cooldown ends, but I'll get back to you on that.

Reply:
oh, that option is indeed valid, but i doubt it'd be that good when you're trying to focus on battle and you have to suddenly click an icon which then makes you look at the chat, completely making you lose focus on the actual game, which is just beyond painful imo

Reply:
So you will need to do this

```
/datum/action/item_action/nanoforgedkatana/New(Target)
   ..()
   RegisterSignal(src, COMSIG_CD_STOP(cleaver_cooldown), .proc/cooldown_end_notif)
```

You will need to make ```/datum/action/item_action/nanoforgedkatana/proc/cooldown_end_notif()``` but that should get you a notification when cooldown ends, in lieu of the cooldown timer overlays spells have.

 This will allow you to run a proc when the cooldown ends to notify that the katana is ready.

Reply:
aaah yeah that's a good idea, and use the span class to make it appear as very big and red in chat so as to not miss it.
Good compromise, yeah! I'll test it out and report back